### PR TITLE
Fixed volume mount issue(759) for Windows

### PR DESCRIPTION
**Pull request by click2cloud-lamda**
### Fixed issue #759 :
- volume binding issue in Windows containers
```
D:\GolandProjects\github.com\nerdctl2\cmd\nerdctl>nerdctl run -it -v C:\ProgramData\nerdctl\052055e3\volumes\default\test\_data:c:\src:rw,rprivate mcr.microsoft.com/windows/nanoserver:20H2
Microsoft Windows [Version 10.0.19042.1586]
(c) 2019 Microsoft Corporation. All rights reserved.

C:\Windows\system32>cd ../..

C:\>dir
 Volume in drive C has no label.
 Volume Serial Number is 7643-AF54

 Directory of C:\

03/03/2022  01:27 PM             5,510 License.txt
03/16/2022  07:00 PM    <DIR>          src
03/16/2022  07:50 PM    <DIR>          Users
03/16/2022  07:50 PM    <DIR>          Windows
               1 File(s)          5,510 bytes
               3 Dir(s)  21,302,005,760 bytes free

C:\>
 ```
When using nerdctl + containerd on Windows, I am able to mount volumes or bind mount to a new windows container and are also able to use bind propagation.

Signed-off-by: Raymond Mathew click2cloud-lamda@click2cloud.net

### CODE REVIEW

**Comment by jsturtevant** [comment]
The windows build failure looks like it was maybe a flake in cirrus?

```
error obtaining VCS status: exec: "git": executable file not found in %PATH%
	Use -buildvcs=false to disable VCS stamping.
```

File: cmd/nerdctl/run_mount_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	base.Cmd("run",
+		"-d",
+		"--name", containerName,
+		"-v", fmt.Sprintf("%s:C:\\mnt1", rwDir),
+		"-v", fmt.Sprintf("%s:C:\\mnt2ro", roDir),
+		"-v", fmt.Sprintf("%s:C:\\mnt3", rwVolName),
+		"-v", fmt.Sprintf("%s:C:\\mnt4ro", roVolName),
+		testutil.WindowsNano).AssertOK()
+	base.Cmd("container", "inspect", containerName).AssertOK()
```
Comment (jsturtevant) [inline cmd/nerdctl/run_mount_windows_test.go:56]: This validates the container started ok but should we validate the mounts are in the right place?

File: cmd/nerdctl/run_mount_windows_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		base.Cmd("volume", "create", v).AssertOK()
+	}
+
+	containerName := tID
+	defer base.Cmd("rm", "-f", containerName).Run()
+
+	base.Cmd("run",
+		"-d",
+		"--name", containerName,
+		"-v", fmt.Sprintf("%s:C:\\mnt1", rwDir),
```
Comment (jsturtevant) [inline cmd/nerdctl/run_mount_windows_test.go:51]: Is there a test for the left side having `c:\`?

File: pkg/mountutil/mountutil.go [hunk 3]
```diff
@@ -55,68 +55,144 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore) (*Processed, error
...
 		}
-		res.Opts = append(res.Opts, specOpts...)
-	default:
-		return nil, fmt.Errorf("failed to parse %q", s)
 	}
-
 	fstype := "nullfs"
 	if runtime.GOOS != "freebsd" {
-		fstype = "none"
+		fstype = ""
```
Comment (jsturtevant) [inline pkg/mountutil/mountutil.go:195]: why is this being dropped?

File: pkg/mountutil/mountutil_windows.go [hunk 4]
```diff
@@ -38,31 +41,195 @@ const DefaultPropagationMode = ""
...
+			//       of RootfsPropagation is unix.MS_SLAVE | unix.MS_REC (i.e. runc applies
+			//       "slave" to all mount points in the container recursively). This ends
+			//       up marking the bind src directories "slave" and preventing it to shared
+			//      with the host. So we set RootfsPropagation to "shared" here.
+			//
+			// See also:
+			// - OCI Runtime Spec: https://github.com/opencontainers/runtime-spec/blob/v1.0.2/config-linux.md#rootfs-mount-propagation
+			// - runc implementation: https://github.com/opencontainers/runc/blob/v1.0.0/libcontainer/rootfs_linux.go#L771-L777
+			specOpts = append(specOpts, func(ctx context.Context, cli oci.Client, c *containers.Container, s *oci.Spec) error {
+				switch s.Linux.RootfsPropagation {
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply pkg/mountutil/mountutil_windows.go:161]: Is this copied from somewhere else?

**Comment by AkihiroSuda** [comment]
Please add DCO sign
https://github.com/apps/dco

**Comment by AkihiroSuda** [comment]
> Signed-off-by: click2cloud-lamda

Please use real name

**Pull request closed without merging**